tianshou/data/buffer.py

```diff
@@ -145,16 +145,10 @@ def __repr__(self) -> str:
 
     def __getattr__(self, key: str) -> Union['Batch', Any]:
         """Return self.key"""
+        if key.startswith('__'):  # since we do not use key begin with "__"
```

Comment:
You really have to do this or if is juste in case ?

Reply:
I'm not sure if there is a better way. Any suggestions?

Reply:
No I'm fine with it. See m'y other comment. Batch has both internal ans user defined keys, it is not surprising to restrict the names that can ne defined by the user.

Reply:
> Ok so it should be mentioned in the doc that user keys must not start with `__` (which means hidden in Python anyway so it makes total sense)

But this is buffer, and users cannot define key by themselves (we reserve 7 keys "obs", ...)

Reply:
Hum Indeed m'y bad. So maybe there is a better way. But it isn't too bad.

Reply:
I changed to `key.startswith('__') and key.endswith('__')`, seems better.

Reply:
I tried with 
```python
try:
    return self._meta[key]
except KeyError:
    raise AttributeError
```
but it doesn't work.